packages/legend-application-query/src/stores/QueryBuilderPropertyEditorState.ts

```diff
@@ -67,9 +68,18 @@ export const getDerivedPropertyMilestoningSteoreotype = (
   if (owner instanceof Class) {
     return getMilestoneTemporalStereotype(owner, graph);
   } else if (owner instanceof Association) {
-    if (owner._generatedMilestonedProperties.length) {
+    const generatedMilestonedProeprties =
+      owner._generatedMilestonedProperties.filter(
+        (prop) =>
+          prop.name !== property.name &&
```

Comment:
I think its fair at this point to make this a helper function in DomainHelper. What do you think?

Reply:
yeah